pennylane/transforms/decompositions/clifford_t_transform.py

```diff
@@ -330,12 +331,71 @@ def _merge_param_gates(operations, merge_ops=None):
     return merged_ops, number_ops
 
 
-# pylint: disable=too-many-branches
+@lru_cache(maxsize=2000)
```

Comment:
Could you remind me where this `2000` magic number came from?

Reply:
The underlying decomposition (Solovay-Kitaev or even the newer ones currently being implemented) returns the decomposition using ~5 unique operations (H, S, T, Sadj, Tadj) or (H, S, T). So the `2000` essentially translates to this working with at least on average `500` wires. Ideally, this can be made tunable if we see that users want to use it.

Reply:
Thanks! (optional) I'd add a code comment here.